docs/fhir-hl7v2-converter/configuration.md

```diff
@@ -0,0 +1,167 @@
+5. value and hl7Spec - if the element is to set a value
+6. schema and resourceIndex - if the element is a child schema
+
+# About Constants
+Constants provide great flexibility to generate reusable schemas and handle repeating types and fields, but with great 
+flexibility can also come great headaches and care must be taken on how constants are used.
+
+1. Care must be taken not to use FHIR Path syntax in an HL7 v2 location spec. It is recommended to use specific constants
+for FHIR Path and others for Hl7 v2 location specs to reduce the risk of a syntax error.
+2. Simple integer constants, like a resource index, can be used in both FHIR Path expressions and HL7 Specs
+(e.g. `myIndex: 1`)
+3. Constants are treated as strings until they are used on a specific element. At that time, if a constant is used
+in a FHIR Path expression then the contents of the string are evaluated as a FHIR Path expression for the result to 
+be inserted as replacement for the constant. This allows you to define constants with FHIR Path expressions
+that can be reused, but that the result will be dependent on what constants and focus resource is set on the element using it.
+4. Constants used as literals in FHIR Path expressions need to be surrounded by quotes (e.g. `myConstant: '"my literal string"'`)
+
+Constants are used as follows:
+* **For FHIR Path expressions** please refer to the
+[FHIR Path specification section on environment variables](http://hl7.org/fhirpath/N1/#environment-variables).
+Constants must have proper FHIR Path expression syntax including for string literals. This library also supports
+having constants as a prefix by using the syntax `` %`<constant_name>-<prefix> ``
+```yaml
+  constants:
+    diagnostic: 'Bundle.entry.resource.ofType(DiagnosticReport)[%orderIndex]'
+    rsext: '"https://reportstream.cdc.gov/fhir/StructureDefinition/"'
+  resource: '%resource.extension(%`rsext-parent-observation-value-descriptor`)'
+```
+
+* **For HL7 location spec**, all constants are treated as string replacements. The special syntax of
+`%{<constant_name>}` is used 
+```yaml
+  constants:
+     fieldPath: '/PATIENT_RESULT/ORDER_OBSERVATION(%{orderIndex})/OBR'
+  hl7Spec: [ '%{fieldPath}(%{entityIdIndex})-3' ]
+```
+
+# Extending Schemas
+Top level schemas can extend and existing top level schema in order to reduce duplicate elements and facilitate
+customization for different purposes. You can chain as many top level schemas as you wish. Note that reusing an element
+name will result in overwriting the existing element. 
+
+```yaml
+hl7Type: ORU_R01
+hl7Version: 2.5.1
+extends: ../ORU_R01/ORU_R01-base
+elements:
+  # Disable an element. Note we only need the name and what is overwritten.
+  - name: filler-order
+    condition: 'false'
+```
+
+Extended schemas can:
+* Define a different type and version of Hl7 v2 message than the top level schema
+* Override one or more elements based on the element name
```

Comment:
We go back and forth between `overwrite` and `override` a bit, we might want to go through and make it consistent throughout, though I don't have a strong opinion of which of the two to use.

Reply:
Yeah. I did that change and did not go through to make sure everything is the same. Will update